voc/python/ast.py

```diff
@@ -2260,15 +2260,15 @@ def visit_Bytes(self, node):
         self.context.add_opcodes(
             java.New('org/python/types/Bytes'),
 
-            JavaOpcodes.BIPUSH(len(node.s)),
+            ICONST_val(len(node.s)),
             JavaOpcodes.NEWARRAY(JavaOpcodes.NEWARRAY.T_BYTE),
         )
 
         for i, b in enumerate(node.s):
             self.context.add_opcodes(
                 JavaOpcodes.DUP(),
                 ICONST_val(i),
-                JavaOpcodes.BIPUSH(node.s[i]),
+                JavaOpcodes.BIPUSH(b if b <= 127 else b - 256),
```

Comment:
Not sure I'm following the purpose of this. At the very least, a clarifying code comment would be helfpul.

Reply:
'b' is an _unsigned_ 8-bit integer, but BIPUSH expects a _signed_ 8-bit integer. I will see about adding a comment to that effect.

Reply:
I added a comment at this point in the code, as well as provided a bit more information in the commit message.

Reply:
Is there a test of this specific case? It's not obvious to me how this "else" branch would manifest in code...

Reply:
Part of fixing for long b'' literals (which I did add a test for) involves changing JavaOpcodes.BIPUSH() from accepting values [0 - 255] to accepting values [-128 - 127].  Without this corresponding change, several _existing_ tests in tests.datatypes.test_bytes.BytesTests fail with "AssertionError: byte format requires -128 <= number <= 127".